k2vr-installer-gui/Tools/Utils.cs

```diff
@@ -20,40 +20,57 @@ public static bool IsDirectoryEmpty(string path)
         public static bool EnsureSteamVrClosed()
         {
             Logger.Log("Checking if SteamVR is open...", false);
-            foreach (Process process in Process.GetProcesses())
+
+            if (Process.GetProcesses().FirstOrDefault((Process proc) => proc.ProcessName == "vrserver" || proc.ProcessName == "vrserver") != null)
             {
-                if (process.ProcessName == "vrmonitor")
+                //Ask if SteamVR should be closed
+                if (MessageBox.Show(
+                    "SteamVR needs to be closed to continue the installation." + Environment.NewLine + "Do you want Setup to close SteamVR?",
+                    "Close SteamVR?",
+                    MessageBoxButton.YesNo,
+                    MessageBoxImage.Warning
+                    ) != MessageBoxResult.Yes)
                 {
-                    Logger.Log("Closing vrmonitor...", false);
-                    process.CloseMainWindow();
-                    Thread.Sleep(5000);
-                    if (!process.HasExited)
+                    //Only exit if SteamVR is still running
+                    if (Process.GetProcesses().FirstOrDefault((Process proc) => proc.ProcessName == "vrserver" || proc.ProcessName == "vrserver") != null)
```

Comment:
Same here. Also UX of this is a problem. If I close SteamVR by hand, and then click "No" (thought process: I already closed it myself, I don't need this tool to do it for me), it will still close the installer.